src/visualizations/config/adapters/dhis_highcharts/legend.js

```diff
@@ -107,27 +111,79 @@ const formatLabel = ({
     result.push(
         '<div style="display: flex; align-items: center; margin-bottom: 4px;" class="data-test-series-key-item">'
     )
-    if (legendSet?.legends?.length && isLegendSetType(visType)) {
-        legendSet.legends.forEach((legend, index) =>
-            result.push(
-                `<span style="${getBulletStyleByFontStyle(
-                    fontStyle
-                )} background-color: ${
-                    legend.color
-                }; margin-right:-5px; z-index: ${
-                    legendSet.legends.length - index
-                }" class="data-test-series-key-item-bullet"></span>`
-            )
+    if (
+        (!seriesId || seriesId.startsWith('trendline')) &&
+        seriesType === getType(VIS_TYPE_LINE).type
+    ) {
+        // trendline
+        result.push(
+            `<span style="height: ${
+                fontStyle[FONT_STYLE_OPTION_FONT_SIZE] / 6.5
+            }px; width: ${
+                fontStyle[FONT_STYLE_OPTION_FONT_SIZE]
+            }px; background-color: ${seriesColor}; display: inline-block;"></span>`
         )
         result.push(
             `<span style="margin-left: 8px" class="data-test-series-key-item-name">${seriesName}</span>`
         )
-    } else {
+    } else if (
+        legendSet?.legends?.length &&
+        isLegendSetType(visType) &&
+        seriesType !== getType(VIS_TYPE_LINE).type
+    ) {
+        // item with legend set
+        legendSet.legends
+            .sort((a, b) => a.startValue - b.startValue)
+            .forEach((legend, index) =>
+                result.push(
+                    `<span style="${getBulletStyleByFontStyle(
+                        fontStyle
+                    )} background-color: ${
+                        legend.color
+                    }; margin-right:-5px; z-index: ${
+                        legendSet.legends.length - index
+                    }" class="data-test-series-key-item-bullet"></span>`
+                )
+            )
         result.push(
-            `<span style="${getBulletStyleByFontStyle(
-                fontStyle
-            )} background-color: ${seriesColor}; margin-right:5px" class="data-test-series-key-item-bullet"></span>`
+            `<span style="margin-left: 8px" class="data-test-series-key-item-name">${seriesName}</span>`
         )
+    } else {
+        // regular item (not a trendline, no applied legend set)
+        const hasPattern = seriesColor?.patternIndex !== undefined
```

Comment:
Is this variable used anywhere else? Perhaps drop it if not.

Reply:
Good point, it was used in more places before, but now its unnecessary